packages/grid/_modules_/grid/hooks/features/sorting/gridSortingUtils.ts

```diff
@@ -39,6 +39,7 @@ const parseSortItem = (sortItem: GridSortItem, apiRef: GridApiRef): GridParsedSo
   const getSortCellParams = (id: GridRowId): GridSortCellParams => ({
     id,
     field: column.field,
+    rowNode: apiRef.current.getRowNode(id)!,
```

Comment:
I added the `rowNode` property to `GridSortCellParams` to be able to base the sorting on the `groupingField` of the row.